src/vs/code/electron-main/app.ts

```diff
@@ -815,6 +819,29 @@ export class CodeApplication extends Disposable {
 		if (updateService instanceof Win32UpdateService || updateService instanceof LinuxUpdateService || updateService instanceof DarwinUpdateService) {
 			updateService.initialize();
 		}
+
+		// If enable-crash-reporter argv is undefined then this is a fresh start,
+		// based on telemetry.enableCrashreporter settings, generate a UUID which
+		// will be used as crash reporter id and also update the json file.
+		const fileService = accessor.get(IFileService);
+		const argvContent = await fileService.readFile(this.environmentService.argvResource);
+		const argvString = argvContent.value.toString();
+		const argvJSON = JSON.parse(stripComments(argvString));
+		if (argvJSON['enable-crash-reporter'] === undefined) {
+			const enableCrashReporter = this.configurationService.getValue<boolean>('telemetry.enableCrashReporter');
```

Comment:
this returns `undefined` in the main process. Is it too early to check the value ?

Reply:
@deepak1556 configuration service in `electron-main` has no defaults because they get registered in renderer only. @sandy081 might have an issue for that, but for now we have to fill in the default value manually.

Reply:
Yeah this is the limitation of using configuration service in non renderer process (main, shared). There could be an issue exists already but not sure.